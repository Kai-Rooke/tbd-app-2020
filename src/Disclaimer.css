.disclaimer{
    width:500px;
    height:450px;
    align-content: center;
    background-color: #31B7BC;
    color: black;
padding-left: 30px; padding-right: 30px;
    border:1px solid #005C6E;
    border-radius: 25px;
}

.text{
    color: black;


}

.text:hover{

    color: black    
}