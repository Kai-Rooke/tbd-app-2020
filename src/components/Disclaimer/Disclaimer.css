.disclaimer{
    width:50%;
    align-content: center;
    background-color: #31B7BC;
    color: white;
    padding-left: 30px; padding-right: 30px;
    border:1px solid #005C6E;
    border-radius: 25px;
    font-family: 'Montserrat', sans-serif;
    font-size: 22px;}


.text{
    color: white;
    text-decoration:none;
    font-family: 'Montserrat', sans-serif;
    font-size: 22px;
}

.text:hover{

    color: white; 
    text-decoration:none;
    font-family: 'Montserrat', sans-serif;
    font-size: 22px;
}
