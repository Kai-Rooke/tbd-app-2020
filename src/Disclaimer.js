import React from 'react';
import './Disclaimer.css';
function Disclaimer() {
    return (
        <div className="Disclaimer">

<br></br>
<form action="https://canvas.anglia.ac.uk/courses/17176/modules">
<div className="disclaimer">
    <b>PRIVACY STATEMENT:</b><br></br> â€œWe will only use your personal information to administer your application and / or enrolment and to provide the products and services you have requested from us.<br></br> Further information on our privacy policy can be found at https://www.ucp.ac.uk/privacy-and-cookies/ or by emailing <a href="mailto:hello@ucp.ac.uk" className="text">hello@ucp.ac.uk</a></div>
</form>

</div>
    )
    }

export default Disclaimer;
