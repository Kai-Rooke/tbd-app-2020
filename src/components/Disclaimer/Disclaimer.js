import React from 'react';
import './Disclaimer.css';
import UCPButton from '../Buttons/UCPButton'

    

const staffDisclaimer = {
    title: "Please Read",
    intro: "Hello you're through to the University Centre Peterborough Clearing Team - how can I help?",
    body: "We will only use your personal information to administer your application and / or enrolment and to provide the products and services you have requested from us. Further information on our privacy policy can be found at https://www.ucp.ac.uk/privacy-and-cookies/ or by emailing hello@ucp.ac.uk"
}
const studentDisclaimer = {
    title: "Privacy Statement",
    intro: "",
    body: "We will only use your personal information to administer your application and / or enrolment and to provide the products and services you have requested from us. Further information on our privacy policy can be found at https://www.ucp.ac.uk/privacy-and-cookies/ or by emailing hello@ucp.ac.uk"
}

function Disclaimer(props) {

    function HandelClick(){     
        
        if(document.getElementById('confirmButton') !== undefined){
          document.getElementById('confirmButton').removeAttribute("hidden");
          document.getElementById('acceptButton').setAttribute("hidden", true);
        }
        if(document.getElementById('confirmButton1') !== undefined){
            document.getElementById('confirmButton1').removeAttribute("hidden");
            document.getElementById('confirmButton2').removeAttribute("hidden");
           
          }
    }
    var disclaimer;
    if(global.userType === "student")
    {
        disclaimer = studentDisclaimer; 
    }
    else{
        disclaimer = staffDisclaimer;
    }
    return (
        <div className="Disclaimer">
            <br></br>
                <link href="https://fonts.googleapis.com/css2?family=Montserrat:wght@500;700&display=swap" rel="stylesheet"></link>
                    <form action="https://canvas.anglia.ac.uk/courses/17176/modules">
                        <div className="disclaimer" data-testid="ucpDisclaimer">
                        <p><b>{disclaimer.title}</b><br/><b>{disclaimer.intro}</b><br/>{disclaimer.body}</p> 
                <div id="acceptButton">        
                    <UCPButton primary="True" className="smallbutton" buttonText="Accept" onClick={HandelClick} to="none"/>
                </div>
            </div>
        </form>
    </div>
    )
}

export default Disclaimer;
