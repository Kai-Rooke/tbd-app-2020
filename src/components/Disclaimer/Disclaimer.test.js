import React from 'react';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom';
import Disclaimer from './Disclaimer';

//cleans up 
afterEach(cleanup);

it("Renders Disclaimer", () => {
    const { asFragment } = render(<Disclaimer/>);
    expect(asFragment()).toMatchSnapshot();
});

it("Disclaimer Test Id", () => {
    const { getByTestId } = render(<Disclaimer/>);
    expect(getByTestId('ucpDisclaimer'));
});

it("Correct Text Present", () => {
    const { getByText } = render(<Disclaimer/>);
    expect(getByText('PRIVACY STATEMENT:'));
});