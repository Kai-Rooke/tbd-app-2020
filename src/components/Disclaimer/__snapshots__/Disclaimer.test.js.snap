// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Disclaimer 1`] = `
<DocumentFragment>
  <div
    class="Disclaimer"
  >
    <br />
    <link
      href="https://fonts.googleapis.com/css2?family=Montserrat:wght@500;700&display=swap"
      rel="stylesheet"
    />
    <form
      action="https://canvas.anglia.ac.uk/courses/17176/modules"
    >
      <div
        class="disclaimer"
        data-testid="ucpDisclaimer"
      >
        <p>
            
          <b>
            PRIVACY STATEMENT:
          </b>
          <br />
           We will only use your personal information to administer your application and / or enrolment and to provide the products and services you have requested from us.
          <br />
           Further information on our privacy policy can be found at https://www.ucp.ac.uk/privacy-and-cookies/ or by emailing 
          <a
            class="text"
            href="mailto:hello@ucp.ac.uk"
          >
            hello@ucp.ac.uk
          </a>
        </p>
        <a
          href="/emptyLink"
        >
          <button
            class="smallbutton"
            data-testid="ucpButton"
            primary="True"
            type="button"
          >
            closeâ€‹
          </button>
        </a>
      </div>
    </form>
  </div>
</DocumentFragment>
`;
